File: python/euklides.py
def main():
    a = int(input('Podaj pierwszą liczbę: '))
    b = int(input('Podaj drugą liczbę: '))

    if a == 0 and b == 0:
        print('Największy wspólny dzielnik dla dwóch zer nie istnieje')
        return

    while b != 0 :
        c = a % b
        a, b = b, c

    print(f'Największy wspólny dzielnik to { abs(a) }')


if __name__ == '__main__':
    main()

File: c/euklides.c
#include <stdio.h>

int main() {
    int a, b, c;

    printf("Podaj pierwszą liczbę: ");
    scanf("%d", &a);
    printf("Podaj drugą liczbę: ");
    scanf("%d", &b);

    if (a == 0 && b == 0) {
        printf("Największy wspólny dzielnik z dwóch zer nie istnieje\n");
        return 0;
    }

    while (b != 0) {
        c = a % b;
        a = b;
        b = c;
    }

    if (a < 0) { a = -a; }
    printf("Największy wspólny dzielnik to %d\n", a);

    return 0;
}